tutorials/tts/FastPitch_MultiSpeaker_Pretraining.ipynb

```diff
@@ -0,0 +1,867 @@
+   "outputs": [],
+   "source": [
+    "train_manifest = os.path.abspath(os.path.join(manidir, 'train.json'))\n",
+    "valid_manifest = os.path.abspath(os.path.join(manidir, 'dev.json'))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc0824a5",
+   "metadata": {},
+   "source": [
+    "## b. Preprocessing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ac16425",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# additional files\n",
+    "!mkdir -p {normdir} && cd {normdir} \\\n",
+    "&& wget https://raw.githubusercontent.com/nvidia/NeMo/$BRANCH/scripts/tts_dataset_files/cmudict-0.7b_nv22.10 \\\n",
+    "&& wget https://raw.githubusercontent.com/nvidia/NeMo/$BRANCH/scripts/tts_dataset_files/heteronyms-052722 \\"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29b11129",
+   "metadata": {},
+   "source": [
+    "### Add absoluate audio path in manifest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8c062437",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def json_reader(filename):\n",
+    "    lines = []\n",
+    "    with open(filename) as f:\n",
+    "        for line in f: lines.append(json.loads(line))\n",
+    "    return lines\n",
+    "\n",
+    "def json_writer(manifest, filename):\n",
+    "    with open(filename, 'w') as fout:\n",
+    "        for m in manifest: fout.write(json.dumps(m) + '\\n') "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "586e604a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_datas = json_reader(train_manifest)\n",
+    "for m in train_datas: m['audio_filepath'] = os.path.abspath(os.path.join(manidir, m['audio_filepath']))\n",
+    "json_writer(train_datas, train_manifest)\n",
+    "\n",
+    "valid_datas = json_reader(valid_manifest)\n",
+    "for m in valid_datas: m['audio_filepath'] = os.path.abspath(os.path.join(manidir, m['audio_filepath']))\n",
+    "json_writer(valid_datas, valid_manifest)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "89e10b53",
+   "metadata": {},
+   "source": [
+    "### Calibrate speaker id to start from 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d0a0f23d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_datas = json_reader(train_manifest)\n",
+    "speaker2id = {s: _id for _id, s in enumerate(set([m['speaker'] for m in train_datas]))}\n",
+    "for m in train_datas: m['old_speaker'], m['speaker'] = m['speaker'], speaker2id[m['speaker']]\n",
+    "json_writer(train_datas, train_manifest)\n",
+    "\n",
+    "valid_datas = json_reader(valid_manifest)\n",
+    "for m in valid_datas: m['old_speaker'], m['speaker'] = m['speaker'], speaker2id[m['speaker']]\n",
+    "json_writer(valid_datas, valid_manifest)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14c74cab",
+   "metadata": {},
+   "source": [
+    "### Calculate Pitch Stats"
```

Comment:
Can use script https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/tts/compute_speaker_stats.py

Reply:
This script should be run after `extract_sup_data.py`. And `extract_sup_data.py` is overwhelmed to go through dataset and dataloader.

Reply:
What do you mean "overwhelmed"?

It would be better to run `extra_sup_data.py` before training to avoid effectively duplicating its code in the tutorial, and to avoid computing the dataset pitch twice (pitch calculation is extremely slow). Using `compute_speaker_stats.py` then also gives the user the option of using speaker-level pitch norm instead of global pitch norm (which tends to work better for multi-speaker).

Reply:
We may have this [PR](https://github.com/NVIDIA/NeMo/pull/6317) to compute pitch stats. Do I still use `extract_sup_data.py` currently?

Reply:
Current workflow is `extract_sup_data.py` --> `compute_speaker_stats.py`

After that PR there will be 2 new scripts 'extract_features.py' --> 'compute_feature_stats.py', which do more or less the same thing with slightly different CLI arguments.

Reply:
I see. I'll use `exetract_sup_data.py` in this PR.